hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-1.x-replication/src/test/java/com/google/cloud/bigtable/hbase1_x/replication/HbaseToCloudBigtableReplicationEndpointTest.java

```diff
@@ -450,34 +450,53 @@ public void testReplicationWithScope() throws IOException, InterruptedException
 
     Put put1 = new Put(ROW_KEY);
     put1.addColumn(TestUtils.CF1, TestUtils.COL_QUALIFIER, 0, ROW_KEY);
-    put1.addColumn(TestUtils.CF2, TestUtils.COL_QUALIFIER, 0, ROW_KEY);
+    put1.addColumn(TestUtils.CF2, TestUtils.COL_QUALIFIER_2, 0, ROW_KEY);
     hbaseTable.put(put1);
 
     TestUtils.waitForReplication(
         () -> {
-          //  replicate Entries will be 1
           return TestReplicationEndpoint.replicatedEntries.get() >= 1;
         });
 
     Result cbtResult = cbtTable.get(new Get(ROW_KEY).setMaxVersions());
     Result hbaseResult = hbaseTable.get(new Get(ROW_KEY).setMaxVersions());
+
+    // make sure we have replicated cells
     Assert.assertFalse(cbtResult.isEmpty());
-    Assert.assertFalse(hbaseResult.isEmpty());
-    Assert.assertEquals(
-        "Number of cells , actual cells: " + hbaseResult.listCells(),
-        2,
-        2,
-        hbaseResult.listCells().size());
+    List<Cell> hbaseCells = hbaseResult.listCells();
+    List<Cell> cbtCells = cbtResult.listCells();
+
+    Assert.assertEquals("bigtable cells", 1, cbtCells.size());
+    // make sure that only CF1 is replicated
+    for (int i = 0; i < hbaseCells.size(); i++) {
+      // make sure CF1 is only replicated
+      if (CellUtil.cloneFamily(hbaseCells.get(i)) == CF1) {
```

Comment:
CF1 will always be 0 index as the results are sorted. you can remove the loop if you want.